test/end2end_test.go

```diff
@@ -6855,8 +6868,12 @@ func (s) TestHTTPHeaderFrameErrorHandlingNormalTrailer(t *testing.T) {
 			},
 			errCode: codes.Internal,
 		},
-	} {
-		doHTTPHeaderTest(t, test.errCode, test.responseHeader, test.trailer)
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			doHTTPHeaderTest(t, test.errCode, test.responseHeader, test.trailer)
```

Comment:
Optional: 

Test helpers, as referred to in Go, are functions which perform some setup or cleanup task associated with the test.  See [here](go/go-style/decisions#mark-test-helpers).

But here, `doHTTPHeaderTest()` is the one which runs the actual test, but is marked as test helper with a call to `t.Helper()` and directly calls `t.Error/Fatal` etc. I see that there are bunch of callers for this function (~6). You can either fix this function to not call `t.Helper()` and instead return an error and have the calling test call `t.Error/Fatal`, or add a TODO here and we will get to this at some point in the future.

Reply:
done. please let me know how the changes look like @easwars